homework/index.js

```diff
@@ -30,13 +30,75 @@
     return elem;
   }
 
+  function pageConstructor(data) {
+    data.sort((a, b) => a.name.localeCompare(b.name));
+    const root = document.getElementById('root');
+    const headerDiv = createAndAppend('div', root, { class: 'headerDiv' });
+    createAndAppend('p', headerDiv, { text: 'HYF Repositories' });
+    const headerSelect = createAndAppend('select', headerDiv, { class: 'headerSelect' });
+    data.map((elem, index) =>
+      createAndAppend('option', headerSelect, { text: elem.name, value: index }),
+    );
+    const container = createAndAppend('div', root, { class: 'container' });
+    const repoInfoDiv = createAndAppend('div', container, { class: 'repoInfoDiv' });
+    const ul = createAndAppend('ul', repoInfoDiv);
+    const repository = createAndAppend('li', ul, { text: `Repository :` });
+    createAndAppend('a', repository, {
+      text: `${data[0].name}`,
+      href: data[0].html_url,
+      target: '_blank',
+      id: 'repository',
+    });
+    createAndAppend('li', ul, { text: `Description :${data[0].description}`, id: 'description' });
+    createAndAppend('li', ul, { text: `Forks :${data[0].forks}`, id: 'forks' });
+    createAndAppend('li', ul, { text: `Update :${data[0].updated_at}`, id: 'update' });
+    const contributorDiv = createAndAppend('div', container, { class: 'contributorDiv' });
+    createAndAppend('p', contributorDiv, { text: 'Contributions' });
+    const contributorUl = createAndAppend('ul', contributorDiv, { class: 'contributorUl' });
+    function contributor() {
+      fetchJSON(data[headerSelect.value].contributors_url, (err, dt) => {
+        if (err) {
+          createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        } else {
+          while (contributorUl.firstChild) {
+            contributorUl.removeChild(contributorUl.firstChild);
+          }
+        }
+        for (let i = 0; i < dt.length; i++) {
+          const li = createAndAppend('li', contributorUl);
+          const aForLi = createAndAppend('a', li, { href: dt[i].html_url, target: '_blank' });
+          createAndAppend('img', aForLi, { src: dt[i].avatar_url });
+          createAndAppend('p', aForLi, { text: dt[i].login });
+          createAndAppend('p', aForLi, { text: dt[i].contributions, class: 'contNumber' });
+        }
+      });
+    }
+    contributor();
+    headerSelect.addEventListener('change', event => {
+      const change = data[event.target.value];
```

Comment:
isn't this the repository? 🤔💭

Never mind but just variable naming have to be meaningful to it's content, you did your best but more on that with experience.